File: session-05-chaos-engineering/typescript/src/order/order.controller.ts
import {Controller, Get, NotFoundException, Param, Post} from "@nestjs/common";
import {OrderService} from "./order.service";

@Controller("orders")
export class OrderController {
    constructor(private readonly orderService: OrderService) {
    }

    @Get()
    listOrders() {
        return this.orderService.listOrders();
    }

    @Get(":id")
    getOrder(@Param("id") id: string) {
        const order = this.orderService.findOrder(parseInt(id));
        if (!order) throw new NotFoundException("Order not found");
        return order;
    }

    @Post(":id/prepare")
    prepareOrder(@Param("id") id: string) {
        const order = this.orderService.findOrder(parseInt(id));
        if (!order) throw new NotFoundException("Order not found");
        return this.orderService.placeOrder(order.id, order.drink);
    }
}

File: session-06-capstone/java/src/main/java/coffeeshop/order/OrderController.java
package coffeeshop.order;

import coffeeshop.kitchen.Preparation;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/orders")
public class OrderController {

    private final OrderService orderService;

    public OrderController(OrderService orderService) {
        this.orderService = orderService;
    }

    @GetMapping
    public List<Order> listOrders() {
        return orderService.listOrders();
    }

    @GetMapping("/{id}")
    public ResponseEntity<Order> getOrder(@PathVariable int id) {
        return orderService.findOrder(id)
            .map(ResponseEntity::ok)
            .orElse(ResponseEntity.notFound().build());
    }

    @PostMapping("/{id}/prepare")
    public ResponseEntity<Preparation> prepareOrder(@PathVariable int id) {
        return orderService.findOrder(id)
            .map(order -> ResponseEntity.ok(orderService.placeOrder(order.id(), order.drink())))
            .orElse(ResponseEntity.notFound().build());
    }
}
